torch/optim/lr_scheduler.py

```diff
@@ -1496,7 +1497,7 @@ def __init__(
             for lr, group in zip(base_lrs, optimizer.param_groups):
                 if isinstance(group["lr"], Tensor):
                     lr_val = lr.item() if isinstance(lr, Tensor) else lr
-                    group["lr"].fill_(lr)
+                    group["lr"].fill_(lr_val)
```

Comment:
btw, wouldn't `fill_` directly accept scalar tensor values? if so, maybe `.item()` not needed? or is the problem with device mixing (`cuda_tensor.fill_(scalar_cpu_tensor)` or `cpu_tensor.fill_(scalar_cuda_tensor)`)?

Reply:
I think its probably because of the device mixing problem. `.item()` could solve the problem if the `lr` tensor is in devices other than `cpu`.

Reply:
Maybe then it can be replaced as `group["lr"].fill_(lr.to(device = group["lr"].device) if isinstance(lr, Tensor) else lr)` making the device-mixing problem more explicit

Reply:
I think before fixing this, the second question mentioned in https://github.com/pytorch/pytorch/issues/126854#issuecomment-2132123607 needs to be settled? As this question also covers other parameters, such as `max_lr` in `OneCycleLR`, and your solution could be generally applied if the answer to that question is yes.

Reply:
It appears that scalar tensor device mixing works fine with `fill_` - probably `fill_` does a copy under the hood on itself:
```python
import torch
cpu_tensor = torch.tensor(2, device = 'cpu')
gpu_tensor = torch.tensor(1, device = 'cuda')
gpu_tensor.fill_(cpu_tensor)
cpu_tensor.fill_(gpu_tensor)
# completes fine
```

so maybe even `group["lr"].fill_(lr)` could be done without any if checks

Reply:
weird, I thought `fill_` would be like other operations, require two tensors located in the same device.
If this is approved, I'd open another PR for this change~